### DISCUSSION TITLE
Docker, UI not loading on Raspberry PI3B+

### DISCUSSION THREAD

**Discussion by SAS-Malko** [zwave-js/zwave-js-ui#3433] [Q&A] [answered]
Hello,
I think i make a mistake but i can't connect to the webserver:8091
Docker and docker-compose are well installed on my RPI pi3B+
I launch "docker-compose up -d --build" without any error message but i can't connect to the webserver on port 8091 in order to finalize the configuration.
"docker-compose ps" command displays hereunder result :
``` 
Name                  Command               State                                         Ports                                       
-----------------------------------------------------------------------------------------------------------------------------------------
zwave-js-ui   docker-entrypoint.sh node  ...   Up      0.0.0.0:3000->3000/tcp,:::3000->3000/tcp, 0.0.0.0:8091->8091/tcp,:::8091->8091/tcp
```

Is it normal that the ip adress is 0.0.0.0?
Thank you in advance for help
LB

**Comment by robertsLando** [comment]
Could you paste your docker-compose file here?

**Reply by SAS-Malko** [reply]
Hello,

hereunder my docker-compose file
I commented some tests i made. always the same result

`
version: '3.7'
services:
  zwave-js-ui:
    container_name: zwave-js-ui
    image: zwavejs/zwave-js-ui:latest
    restart: always
    tty: true
    stop_signal: SIGINT
    environment:
      #- SESSION_SECRET=mysupersecretkey (tested)
      - SESSION_SECRET=[removed]
      - ZWAVEJS_EXTERNAL_CONFIG=/usr/src/app/store/.config-db
      # Uncomment if you want logs time and dates to match your timezone instead of UTC
      # Available at https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
      #- TZ=America/New_York
    networks:
      - zwave
    devices:
      # Do not use /dev/ttyUSBX serial devices, as those mappings can change over time.
      # Instead, use the /dev/serial/by-id/X serial device for your Z-Wave stick.
      #- '/dev/serial/by-id/insert_stick_reference_here:/dev/zwave'
      #- '/dev/ttyACM0:/dev/zwave' (tested)
      #- '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave' (tested)
      - '/dev/serial/by-id/usb-0658_0200-if00:/dev/ttyACM0'
    volumes:
      - zwave-config:/usr/src/app/store
    ports:
      # port for web interface
      - '8091:8091'
      # port for Z-Wave JS websocket server
      - '3000:3000'
networks:
  zwave:
volumes:
  zwave-config:
    name: zwave-config

`
i looked at the docker-compose.yml in some tutos and it seems the same
I don't know other command to verify that "docker ps"
Thank you for your help

LB

**Reply by SAS-Malko** [reply]
Grrrr indentation losts
`version: '3.7'
services:
  zwave-js-ui:
    container_name: zwave-js-ui
    image: zwavejs/zwave-js-ui:latest
    restart: always
    tty: true
    stop_signal: SIGINT
    environment:
      #- SESSION_SECRET=mysupersecretkey (tested)
      - SESSION_SECRET=[removed]
      - ZWAVEJS_EXTERNAL_CONFIG=/usr/src/app/store/.config-db
      # Uncomment if you want logs time and dates to match your timezone instead of UTC
      # Available at https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
      #- TZ=America/New_York
    networks:
      - zwave
    devices:
      # Do not use /dev/ttyUSBX serial devices, as those mappings can change over time.
      # Instead, use the /dev/serial/by-id/X serial device for your Z-Wave stick.
      #- '/dev/serial/by-id/insert_stick_reference_here:/dev/zwave'
      #- '/dev/ttyACM0:/dev/zwave' (tested)
      #- '/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave' (tested)
      - '/dev/serial/by-id/usb-0658_0200-if00:/dev/ttyACM0'
    volumes:
      - zwave-config:/usr/src/app/store
    ports:
      # port for web interface
      - '8091:8091'
      # port for Z-Wave JS websocket server
      - '3000:3000'
networks:
  zwave:
volumes:
  zwave-config:
    name: zwave-config
`

**Reply by SAS-Malko** [reply]
So i attach the file
[docker-compose.yml.zip](https://github.com/zwave-js/zwave-js-ui/files/13431754/docker-compose.yml.zip)
hope it's well zipped
thanks 
LB

**Reply by robertsLando** [reply]
OK seems ok. 

Please make a [application log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=application-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field).

If you cannot reach the ui simply start docker-compose without the `-d` command and paste here the output

**Reply by SAS-Malko** [reply]
Hello,
As i can't launch UI (i understand it's my problem because i can't join http://ip:8091) i did some commands and send you the results
1/ docker-compose ps
2/ docker-compose stop
3/ docker-compose up 
4/ docker-compose PS
I never launched "docker-compose up" (without -d) before and finally i can see that there is an error!
But i don't know what to do to solve the problem :(

`pi@raspberrypi:~/zwavejs2mqtt $ docker-compose ps
   Name                  Command                 State      Ports
-----------------------------------------------------------------
zwave-js-ui   docker-entrypoint.sh node  ...   Restarting        
pi@raspberrypi:~/zwavejs2mqtt $ docker-compose stop
Stopping zwave-js-ui ... done
pi@raspberrypi:~/zwavejs2mqtt $ docker-compose up
Starting zwave-js-ui ... done
Attaching to zwave-js-ui
zwave-js-ui    | 
zwave-js-ui    | 
zwave-js-ui    | #
zwave-js-ui    | # Fatal error in , line 0
zwave-js-ui    | # unreachable code
zwave-js-ui    | #
zwave-js-ui    | #
zwave-js-ui    | #
zwave-js-ui    | #FailureMessage Object: 0x7e912464
pi@raspberrypi:~/zwavejs2mqtt $ docker-compose ps
   Name                  Command               State                                         Ports                                       
-----------------------------------------------------------------------------------------------------------------------------------------
zwave-js-ui   docker-entrypoint.sh node  ...   Up      0.0.0.0:3000->3000/tcp,:::3000->3000/tcp, 0.0.0.0:8091->8091/tcp,:::8091->8091/tcp
pi@raspberrypi:~/zwavejs2mqtt $ 
`

Thank you
LB

**Reply by robertsLando** [reply, marked as answer]
Ok see https://github.com/zwave-js/zwave-js-ui/issues/2725
